packages/nextjs/src/performance/client.ts

```diff
@@ -0,0 +1,27 @@
+import { TransactionContext } from '@sentry/types';
+
+import * as Sentry from '../index.client';
+import { wrapRouter } from './routerWrapper';
+
+export function startClientPerfMonitoring(): void {
+  wrapRouter();
+  startInitialTransaction();
+}
+
+function startInitialTransaction(): void {
+  const initialTransactionCtx = getInitialTransactionContext();
+  const transaction = Sentry.startTransaction(initialTransactionCtx);
```

Comment:
This won't start an idle transaction, nor use `BrowserTracing` to track requests/metrics. We'll have to move this to routing instrumentation + browser tracing to take advantage of that.

```ts
// sentry.client.config.js
Sentry.init(
  dsn: '...'
  tracesSampleRate: 1.0,
  integrations: [
    new Integrations.BrowserTracing({
      routingInstrumentation: nextRouterInstrumentation,
    }),
  ],
);
```

```ts
let stashedStartTransaction: (context: TransactionContext) => Transaction | undefined;

export function nextRouterInstrumentation(
  customStartTransaction: (context: TransactionContext) => Transaction | undefined,
  startTransactionOnPageLoad: boolean = true,
  startTransactionOnLocationChange: boolean = true,
): void {
  // we can also wrap earlier
  wrapRouter(customStartTransaction);
  if (startTransactionOnPageLoad) {
    customStartTransaction({
      name: window.location.pathname,
      op: 'pageload',
    });
  }
}

export function wrapRouter(customStartTransaction): void {
  // fill stuff and paramaterize names accordingly
} 
```